fix(dev-docs): Buffer fixes for logs and metrics (#15513)
getsentry/sentry-docs: develop-docs/sdk/telemetry/telemetry-buffer/index.mdx

# Common Requirements

This section covers the common requirements relevant for all platforms.

## Data Forwarding Scenarios

> The TelemetryBuffer MUST forward all data in memory to the transport to avoid data loss in the following scenarios:

Reviewer: ```suggestion
The TelemetryBuffer **MUST** forward all data in memory to the transport to avoid data loss in the following scenarios:
```

> 1. When the user calls `SentrySDK.flush()`, the TelemetryBuffer MUST forward all data in memory to the transport, and only then the transport SHOULD flush the data.

Reviewer: ```suggestion
1. When the user calls `SentrySDK.flush()`, the TelemetryBuffer **MUST** forward all data in memory to the transport, and only then **SHOULD**  the transport flush the data.
```

Scenarios mostly relevant for mobile SDKs:

1. When the application shuts down gracefully, the TelemetryBuffer SHOULD forward all data in memory to the transport. The transport SHOULD keep its existing behavior, which usually stores the data to disk as an envelope. It is not required to call transport `flush`. This is mostly relevant for mobile SDKs already subscribed to these hooks, such as [applicationWillTerminate](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/applicationwillterminate(_:)) on iOS.
2. When the application moves to the background, the TelemetryBuffer SHOULD forward all data in memory to the transport and stop the timer. The transport SHOULD keep its existing behavior, which usually stores the data to disk as an envelope. It is not required to call transport `flush`. This is mostly relevant for mobile SDKs.
3. Mobile SDKs MUST minimize data loss when sudden process terminations occur. Refer to the [Mobile Telemetry Buffer](/sdk/telemetry/telemetry-buffer/mobile-telemetry-buffer) section for more details.

## FAQ

### Where is the batch processor?

> The batch processor is deprecated and we move it to the [batch-processor](/sdk/telemetry/telemetry-buffer/batch-processor/) page. The telemetry buffer will include parts of the batch processor functionality.

Reviewer: ```suggestion
The batch processor is deprecated, so we moved it to the [batch-processor](/sdk/telemetry/telemetry-buffer/batch-processor/) page. The telemetry buffer will include parts of the batch processor functionality.
```
